tests/Doctrine/Tests/Common/Cache/FileCacheTest.php

```diff
@@ -129,4 +129,50 @@ public function testGetExtensionReturnsExtensionString()
 
         $this->assertEquals($extension, $actualExtension);
     }
+
+    /**
+     * @runInSeparateProcess
+     *
+     * @covers \Doctrine\Common\Cache\FileCache::getFilename
+     */
+    public function testWindowsPathLengthLimitationsAreCorrectlyRespected()
+    {
+        if (! defined('PHP_WINDOWS_VERSION_BUILD')) {
+            define('PHP_WINDOWS_VERSION_BUILD', 'Yes, this is the "usual suspect", with the usual limitations');
+        }
+
+        $fileCache = $this->getMockForAbstractClass(
+            'Doctrine\Common\Cache\FileCache',
+            array(__DIR__, '.doctrine.cache')
+        );
+
+
+        $baseDirLength        = strlen(__DIR__);
+        $extensionLength      = strlen('.doctrine.cache');
+        $windowsPathMaxLength = 260;
+        $maxKeyLength         = $windowsPathMaxLength - ($baseDirLength + $extensionLength);
+
+        self::assertSame('61', bin2hex('a'), '(added just for clarity and system integrity check)');
+
+        $tooLongKey = str_repeat('a', ($maxKeyLength / 2) + 1);
+        $fittingKey = str_repeat('a', ($maxKeyLength / 2) - 2); // note: 2 chars due to path separator added as well
+
+        $tooLongKeyHash = hash('sha256', $tooLongKey);
+        $fittingKeyHash = hash('sha256', $fittingKey);
+
+        $getFileName = new \ReflectionMethod($fileCache, 'getFilename');
+
+        $getFileName->setAccessible(true);
+
+        $this->assertSame(
+            __DIR__ . '/' . substr($tooLongKeyHash, 0, 2) . '/_' . $tooLongKeyHash . '.doctrine.cache',
+            $getFileName->invoke($fileCache, $tooLongKey),
+            'Keys over the limit of the allowed length are hashed correctly'
+        );
+        $this->assertSame(
```

Comment:
Meh, needs some hardening around this assertion (fails on travis).

The annoying bit is that the number of chars in a hash is always even, while the sum of extension + base-path can be odd: fun!